components/codec/src/byte.rs

```diff
@@ -345,16 +377,16 @@ impl MemComparableCodecHelper for DescendingMemComparableCodecHelper {
 }
 
 pub trait MemComparableByteEncoder: NumberEncoder {
-    /// write all bytes in ascending memory-comparable format.
+    /// Writes all bytes in ascending memory-comparable format.
     ///
     /// # Errors
     ///
-    /// Returns `Error::BufferTooSmall` if buffer remaining size < values.len().
+    /// Returns `Error::BufferTooSmall` if buffer remaining size is not enough.
```

Comment:
Returns `UnexpectedEof` instead of `BufferTooSmall`, the comment should be updated.